pkg/convert/xliff12.go

```diff
@@ -95,3 +109,59 @@ func ToXliff12(messages model.Messages) ([]byte, error) {
 
 	return append([]byte(xml.Header), data...), nil
 }
+
+// helpers
+
+// positionsFromXliff12 extracts line positions from []contextGroup.
+func positionsFromXliff12(contextGroups []contextGroup) model.Positions {
+	var positions model.Positions
+
+	for _, cg := range contextGroups {
+		if cg.Purpose == "location" {
+			var pos strings.Builder
+
+			for _, c := range cg.Contexts {
+				switch c.Type {
+				case "sourcefile":
+					if pos.Len() > 0 {
+						pos.WriteString(", " + c.Content)
+					} else {
+						pos.WriteString(c.Content)
+					}
+				case "linenumber":
+					pos.WriteString(":" + c.Content)
+				}
+			}
+
+			positions = append(positions, strings.Split(pos.String(), ", ")...)
```

Comment:
Inefficient. Do not build string for all positions and then split!

Create a value for each position and then append it instead.